### PR TITLE
Timeout the dropTable operation

### PR DESCRIPTION
**Pull request by hsaraogi**
**Goals (and why)**: Internal build number 2589 failed due to 
```
java.lang.RuntimeException: The current lock holder has failed to update its heartbeat. We suspect that this might be due to a node crashing while holding the schema mutation lock. If this is indeed the case, run the clean-cass-locks-state cli command.
```
after running for over an hour.

**Implementation Description (bullets)**:
1. dropTables waits at most 5 minutes now or just throws.

**Concerns (what feedback would you like?)**: Is the 5 minute time reasonable given the schema mutation lock heartbeats?

**Where should we start reviewing?**: `AbstractSweeperTest.java`

**Priority (whenever / two weeks / yesterday)**: whenever

### CODE REVIEW

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweeperTest.java [hunk 1]
```diff
@@ -127,8 +129,13 @@ protected static void setupTables(KeyValueService kvs) {
     }
 
     private static void tearDownTables(KeyValueService kvs) {
-        kvs.getAllTableNames().stream()
-                .forEach(tableRef -> kvs.dropTable(tableRef));
+        Awaitility.await()
+                .timeout(Duration.FIVE_MINUTES)
```
Comment (gsheasby) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweeperTest.java]: How long does this take in normal circumstances? Is the five minute timeout chosen due to some kind of heartbeat timeout?

**Comment by tpetracca** [comment]
Is the timeout for kvs.droptTable kvs impl specific? Should we just be changing code to ensure it properly times out faster and not wrapping this specific usage of the dropTable?

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweeperTest.java (hunk 1 above)
Comment (hsaraogi) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweeperTest.java]: It takes about an hour if the build is going to fail. @ashrayjain: for SA, the heartbeat is every second, so maybe  we can reduce the 5 minutes to 1 minute?

**Comment by hsaraogi** [comment]
@tpetracca: This flake appears mostly in Cassandra integration tests. I doubt if it will effect any other kvs. Are you suggesting that Cassandra dropTable operations should timeout if they dont finish in a fixed interval(maybe configurable)? Not sure if that is safe given systems can be under different load.

**Comment by tpetracca** [comment]
I guess I'm fine merging this, it's just one of those things that feels like a bandaid of a symptom instead of addressing the underlying issue itself.

I don't think we do, but if we had actual numbers on how long DROP mutations should take in realistically sized clusters we could add a sane timeout to the dropTable call.

**Comment by gsheasby** [comment]
I think the flakes may be _somewhat_ beyond our control - other projects have reported increased flakiness during the workday, when more builds are running and thus more stress is being put on internal circle. If we see flakes during the early morning (between 3am and 9am UK time), then that points more strongly to something being wrong with the tests.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweeperTest.java (hunk 1 above)
Comment (gsheasby) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweeperTest.java]: I meant, if we successfully drop the table, how long does that take?

**Comment by hsaraogi** [comment]
These flakes occur time and again. And there should be changing our testing in any ways if the dropTable is successful in 5 minutes, I have at least seen 5 of them in the last week. I guess they are likely to go away given changes in the build infra, but we cant say that for certainty.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweeperTest.java (hunk 1 above)
Comment (hsaraogi) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweeperTest.java]: I dont know. But I think less than 5 minutes.

**Pull request merged**